src/importexport/musicxml/internal/musicxml/export/exportmusicxml.cpp

```diff
@@ -7954,7 +7954,11 @@ void ExportMusicXml::writeInstrumentChange(const InstrumentChange* instrChange)
 
     for (KeySig* keySig : instrChange->keySigs()) {
         staff_idx_t st = m_score->staffIdx(part);
-        keysig(keySig, part->staff(st)->clef(instrChange->tick()), m_score->staffIdx(part), keySig->visible());
+        Staff* staff = part->staff(st);
```

Comment:
There is another problem here: `st` is the index of the staff in the _score_, but it is passed to a method that expects the index of the staff in the _part_. It seems that we should just pass `0` instead of `st`.